### DISCUSSION TITLE
Limit Different Users to see limited cameras

### DISCUSSION THREAD

**Discussion by shanekuz** [blakeblackshear/frigate#12292] [Ideas]
I think it would be handy if I could create a user account that could only see a specific camera or a limited amount of cameras.  I thought this would be helpful to be able to give neighbors access to see a shared camera but not have access to all of them.  Would also be good to be able to lock them to view only and not PTZ if available and maybe have or not have access to recordings.

**Comment by BobTB** [comment]
Oh yes, this is very  much indeed needed. Why have users if this is not implemented, and every user can see everything is beyond me....

**Reply by hawkeye217** [reply]
This is not a trivial task, but yes, this is a planned feature.

**Reply by BobTB** [reply]
A limited user, which can access (view) selected cameras and does not have access to config seems a reasonably easy task to do. This would cover 95% of use cases. The old how it is called, hm... ah... Zoneminder has this.

**Reply by NickM-27** [reply]
Authentication in general was just added, user roles will come in a future version, users in general were added as a first step for this. This is not a simple feature to add because it is not just hiding things in the UI but also needs to include limiting API access so it is an actual security feature and not just hiding some things in the UI

ZoneMinder is much older than frigate, so the fact that it has support for user roles has no bearing on how easy or complicated it is to add a feature like this.

**Reply by jrzero314** [reply]
BickM-27 
Thank you for your time in implemeting this feature when it comes to security I know it can be a headache.

**Reply by flyersa** [reply]
Would love to see this too. Its pretty much only for hobby home use without this. I want people to just see special camera groups and the recordings of them without any access to anything else, or at least a global read-only user that cannot see config or do changes would be sufficient too

**Reply by smaugfm** [reply]
Hey, 
Is there an estimation when this feature will arrive?

**Reply by hawkeye217** [reply]
No, there is no estimate.

**Reply by jakubsuchybio** [reply]
Hi guys, just wanted to ask if this feature's estimate changed? I would also love this feature for read-only access of subgroups of cameras per user.
My usecase is more complicated, where I have routed 3 locations appart of each other, but inside one frigate instance, because I have coral on this device. So creating another instance with only CPU inference wouldn't be nice 😅.

**Comment by n3ovo** [comment]
I know this is a lot more work .. and I never though of giving frends/neighbors access to just the Cam that covers some of there yard... very very cool thought ... Glad to 2 see the willingness to think of Future UPDATES ... 

ps to the "Frigate programs" A BIG THANKS FOR ALL YOUR WORK!!

Thanks
 John G.

**Comment by hippieua** [comment]
Yes, ever since the read-only role was introduced, I’ve wanted to give neighbors access to view specific cameras or groups of cameras, while keeping cameras from private areas hidden. The ability to restrict access to certain camera groups would be a fantastic feature.

**Reply by hawkeye217** [reply]
This will be supported in a future version of Frigate. You can follow this feature request: https://github.com/blakeblackshear/frigate/issues/6614

**Comment by supp0815** [comment]
Second that! Please add an option to limit certain cameras to certain users. I switched from BI were it was possible.

**Reply by hawkeye217** [reply]
This will be supported in a future version of Frigate. You can follow this feature request: https://github.com/blakeblackshear/frigate/issues/6614
